core/src/main/java/org/jruby/ext/bigdecimal/RubyBigDecimal.java

```diff
@@ -483,7 +485,7 @@ private RubyBigDecimal getVpValueWithPrec(ThreadContext context, IRubyObject val
             }
 
             MathContext mathContext = new MathContext(RubyFloat.DIG + 1, getRoundingMode(context.runtime));
-            return new RubyBigDecimal(context.runtime, new BigDecimal(doubleValue, mathContext));
+            return new RubyBigDecimal(context.runtime, new BigDecimal(value.toString(), mathContext));
```

Comment:
this is interesting, we have the `double doubleValue` and check it's valid but still decide not to use it, in this branch.
would be nice to explain the reasoning (e.g. in a comment) of falling back to `value.to_s` here instead of using the value from RubyFloat

Reply:
the above snippet is take from the test case.
```
  def test_mult_with_float
    assert_kind_of(BigDecimal, BigDecimal("3") * 1.5)
    assert_equal(BigDecimal("64.4"), BigDecimal(1) * 64.4)
  end
```